pkg/work/spoke/controllers/manifestcontroller/manifestwork_reconciler.go

```diff
@@ -77,6 +80,9 @@ func (m *manifestworkReconciler) reconcile(
 		// Add applied status condition
 		manifestCondition.Conditions = append(manifestCondition.Conditions, buildAppliedStatusCondition(result))
 
+		// Add result conditions
+		manifestCondition.Conditions = append(manifestCondition.Conditions, result.Conditions...)
```

Comment:
I think this reader should be used in status controller, the thing is status controller will return faster and periodically, and you can use status update interval to control how frequent the resource will be read.

Reply:
Hmm ok I'll take a look at that